knowledge_base/index-shrink-before-upgrade-consistent-documents.json

```diff
@@ -0,0 +1,4 @@
+{
+  "id": "index-shrink-before-upgrade-consistent-documents",
```

Comment:
Following @mikaylathompson's convention w/ the DOY bug expectation, maybe this should live in the same JSON file as `index-shrink-after-upgrade-consistent-documents`?  Are they two sides of the same coin?  Alternatively, they may actually be different different coins, and their flipside is that for certain versions it's NOT true...

I'm still trying to figure out what the right structure here is, but it feels like we do need the ability to express different scenarios for a given general topic.  In this case, have `knowledge_base/index_shrink_consistent_documents.json` contain both the "happy-path" and "sad-path" expectations for `index-shrink-after-upgrade-consistent-documents` and `index-shrink-before-upgrade-consistent-documents`.

Also curious about your thoughts on this one @gregschohn

Reply:
They are essentially the same test, with the shrink either happening before the upgrade or after followed by our assertions.

I think I like grouping these together, my only drawback is like you mentioned if we find there is a break for one of both of these expectations we would need to add that here as well but I think that is reasonable. 

There is another side of this that may be `index-expand` which would be increasing the shards which I currently think makes sense to have a separate grouping for.

Reply:
Additional thoughts on this are below.  Quick answer - this problem has the potential to explode, especially as we pick up more nuance and stricter interpretations.  I think that decomposing our expectations into reusable building blocks would help.

I like the spirit of grouping that @lewijacn brings up, but doubt that expectations will be able to be grouped over just a single axis.  This is a good spot where tags could come in.  On the other hand, the building blocks from above might also be able to serve as a grouping tag (documents with geospatial boundaries; clusters that were in the process of being upgraded in place when a [new|old] node failed over]).

I would say that my answer to @chelma's question of structure would be to distill expectations to have only a single responsibility.  At that point, you would at least have a major axis that could serve as a primary axis for grouping, even if it had mixin/dependency behaviors from other building blocks.  

Does that make any sense?